src/sources/prometheus/scrape.rs

```diff
@@ -0,0 +1,430 @@
+
+    fn output_type(&self) -> config::DataType {
+        self.config.output_type()
+    }
+
+    fn source_type(&self) -> &'static str {
+        "prometheus_scrape"
+    }
+}
+
+fn prometheus(
+    urls: Vec<http::Uri>,
+    tls: TlsSettings,
+    auth: Option<Auth>,
+    interval: u64,
+    shutdown: ShutdownSignal,
+    out: Pipeline,
+) -> sources::Source {
+    let out = out
+        .sink_map_err(|error| error!(message = "Error sending metric.", %error))
+        .sink_compat();
+    Box::pin(tokio::time::interval(Duration::from_secs(interval))
+        .take_until(shutdown)
+        .map(move |_| stream::iter(urls.clone()))
+        .flatten()
+        .map(move |url| {
+            let client = HttpClient::new(tls.clone()).expect("Building HTTP client failed");
+
+            let mut request = Request::get(&url)
+                .body(Body::empty())
+                .expect("error creating request");
+            if let Some(auth) = &auth {
+                auth.apply(&mut request);
+            }
+
+            let start = Instant::now();
+            client
+                .send(request)
+                .map_err(crate::Error::from)
+                .and_then(|response| async move {
+                    let (header, body) = response.into_parts();
+                    let body = hyper::body::to_bytes(body).await?;
+                    Ok((header, body))
+                })
+                .into_stream()
+                .filter_map(move |response| {
+                    ready(match response {
+                        Ok((header, body)) if header.status == hyper::StatusCode::OK => {
+                            emit!(PrometheusRequestCompleted {
+                                start,
+                                end: Instant::now()
+                            });
+
+                            let byte_size = body.len();
+                            let body = String::from_utf8_lossy(&body);
+
+                            match parser::parse(&body) {
+                                Ok(metrics) => {
+                                    emit!(PrometheusEventReceived {
+                                        byte_size,
+                                        count: metrics.len(),
+                                    });
+                                    Some(stream::iter(metrics).map(Event::Metric).map(Ok))
+                                }
+                                Err(error) => {
+                                    if url.path() == "/" {
+                                        // https://github.com/timberio/vector/pull/3801#issuecomment-700723178
+                                        warn!(
+                                            message = "No path is set on the endpoint and we got a parse error, did you mean to use /metrics? This behavior changed in version 0.11.",
+                                            endpoint = %url
+                                        );
+                            }
```

Comment:
Has the formatting messed up here?

I'm surprised the CI doesn't seem to have caught it.

Reply:
`rustfmt` occasionally gives up in the presence of macros and leaves code un-reformatted. It's quite frustrating.

Reply:
In any case, that code should be unchanged since it's just a rename. I'm not sure why GitHub is showing it as a separate add.